### PR TITLE
Extract mock rippled constructor, unify some request handlers to reduce redundancy

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: test/mock-rippled.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  _.assign(mock, EventEmitter2.prototype);
+
+  var close = mock.close;
+  mock.close = function() {
+    if (mock.expectedRequests !== undefined) {
+      assert(_.every(mock.expectedRequests, function(counter) {
+        return counter === 0;
+      }));
+    }
+    close.bind(mock);
```
Comment (wltsmrz) [inline test/mock-rippled.js]: .call() ?

**Comment by wltsmrz** [comment]
Except for comment, LGTM

File: test/mock-rippled.js (hunk 1 above)
Comment (clark800) [inline test/mock-rippled.js]: Yes thanks

File: test/mock-rippled.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+module.exports = function(port) {
+  var mock = new WebSocketServer({port: port});
+  _.assign(mock, EventEmitter2.prototype);
+
+  var close = mock.close;
+  mock.close = function() {
+    if (mock.expectedRequests !== undefined) {
+      assert(_.every(mock.expectedRequests, function(counter) {
+        return counter === 0;
+      }));
```
Comment (wltsmrz) [inline test/mock-rippled.js]: Maybe add an error message here

Comment (clark800) [inline test/mock-rippled.js]: Will do

**Pull request merged**